x-pack/plugins/ml/public/application/services/ml_api_service/inference.ts

```diff
@@ -0,0 +1,163 @@
+  count: number;
+  time_in_millis: number;
+  current: number;
+  failed: number;
+}
+
+export interface InferenceStatsResponse {
+  count: number;
+  trained_model_stats: Array<{
+    model_id: string;
+    pipeline_count: number;
+    inference_stats: {
+      failure_count: number;
+      inference_count: number;
+      cache_miss_count: number;
+      missing_all_fields_count: number;
+      timestamp: number;
+    };
+    ingest: {
+      total: IngestStats;
+      pipelines: Record<
+        string,
+        IngestStats & {
+          processors: Array<
+            Record<
+              string,
+              {
+                // TODO use type from ingest_pipelines
+                type: string;
+                stats: IngestStats;
+              }
+            >
+          >;
+        }
+      >;
+    };
+  }>;
+}
+
+export type ModelStats = InferenceStatsResponse['trained_model_stats'][number];
+
+/**
+ * Service with APIs calls to perform inference operations.
+ * @param httpService
+ */
+export function inferenceApiProvider(httpService: HttpService) {
+  const apiBasePath = basePath();
+
+  return {
+    /**
+     * Fetches configuration information for a trained inference model.
+     *
+     * @param modelId - Model ID, collection of Model IDs or Model ID pattern.
+     *                  Fetches all In case nothing is provided.
+     * @param params - Optional query params
+     */
+    getInferenceModel(modelId?: string | string[], params?: InferenceQueryParams) {
+      let model = modelId ?? '';
+      if (Array.isArray(modelId)) {
+        model = modelId.join(',');
+      }
+
+      return httpService.http<InferenceConfigResponse>({
+        path: `${apiBasePath}/inference${model && `/${model}`}`,
+        method: 'GET',
+      });
+    },
+
+    /**
+     * Fetches usage information for trained inference models.
+     *
+     * @param modelId - Model ID, collection of Model IDs or Model ID pattern.
+     *                  Fetches all In case nothing is provided.
+     * @param params - Optional query params
+     */
+    getInferenceModelStats(modelId?: string | string[], params?: InferenceStatsQueryParams) {
+      let model = modelId ?? '_all';
+      if (Array.isArray(modelId)) {
+        model = modelId.join(',');
+      }
+
+      return httpService.http<InferenceStatsResponse>({
+        path: `${apiBasePath}/inference/${model}/_stats`,
+        method: 'GET',
+      });
+    },
+
+    /**
+     * Deletes an existing trained inference model.
+     *
+     * @param modelId - Model ID
+     */
+    deleteInferenceModel(modelId: string) {
+      return httpService.http<any>({
+        path: `${apiBasePath}/inference/${modelId}`,
+        method: 'DELETE',
+      });
+    },
+  };
+}
+
+type InferenceApiService = ReturnType<typeof inferenceApiProvider>;
+
+/**
+ * Hooks for accessing {@link InferenceApiService} in React components.
+ */
+export function useInferenceApiService(): InferenceApiService {
```

Comment:
i like the idea of adding hooks to this files, for ease of use throughout the app.